keyboards/rmi_kb/tkl_f13hs/keymaps/via/keymap.c

```diff
@@ -0,0 +1,743 @@
+            *value_data = g_value[esc] & (1 << bs_gui) ? 1 : 0;
+            break;
+        case id_esc_toggle_on_alt:
+            *value_data = g_value[esc] & (1 << bs_alt) ? 1 : 0;
+            break;
+        case id_esc_toggle_on_shift:
+            *value_data = g_value[esc] & (1 << bs_shift) ? 1 : 0;
+            break;
+        case id_esc_toggle_on_layer_0:
+            *value_data = g_value[esc] & (1 << bs_layer_0) ? 1 : 0;
+            break;
+        case id_esc_toggle_on_layer_1:
+            *value_data = g_value[esc] & (1 << bs_layer_1) ? 1 : 0;
+            break;
+        case id_esc_toggle_on_layer_2:
+            *value_data = g_value[esc] & (1 << bs_layer_2) ? 1 : 0;
+            break;
+        case id_esc_toggle_on_layer_3:
+            *value_data = g_value[esc] & (1 << bs_layer_3) ? 1 : 0;
+            break;
+
+        /* ===============================
+           ========== Caps Lock ==========
+           =============================== */
+        case id_caps_lock_toggle_on_num_lock:
+            *value_data = g_value[caps] & (1 << bs_num_lock) ? 1 : 0;
+            break;
+        case id_caps_lock_toggle_on_caps_lock:
+            *value_data = g_value[caps] & (1 << bs_caps_lock) ? 1 : 0;
+            break;
+        case id_caps_lock_toggle_on_scroll_lock:
+            *value_data = g_value[caps] & (1 << bs_scroll_lock) ? 1 : 0;
+            break;
+        case id_caps_lock_toggle_on_ctrl:
+            *value_data = g_value[caps] & (1 << bs_ctrl) ? 1 : 0;
+            break;
+        case id_caps_lock_toggle_on_gui:
+            *value_data = g_value[caps] & (1 << bs_gui) ? 1 : 0;
+            break;
+        case id_caps_lock_toggle_on_alt:
+            *value_data = g_value[caps] & (1 << bs_alt) ? 1 : 0;
+            break;
+        case id_caps_lock_toggle_on_shift:
+            *value_data = g_value[caps] & (1 << bs_shift) ? 1 : 0;
+            break;
+        case id_caps_lock_toggle_on_layer_0:
+            *value_data = g_value[caps] & (1 << bs_layer_0) ? 1 : 0;
+            break;
+        case id_caps_lock_toggle_on_layer_1:
+            *value_data = g_value[caps] & (1 << bs_layer_1) ? 1 : 0;
+            break;
+        case id_caps_lock_toggle_on_layer_2:
+            *value_data = g_value[caps] & (1 << bs_layer_2) ? 1 : 0;
+            break;
+        case id_caps_lock_toggle_on_layer_3:
+            *value_data = g_value[caps] & (1 << bs_layer_3) ? 1 : 0;
+            break;
+    }
+}
+
+void indicator_config_load(void) {
+    eeprom_read_block(&g_value, ((void*)VIA_EEPROM_CUSTOM_CONFIG_ADDR), VIA_EEPROM_CUSTOM_CONFIG_SIZE);
+}
+
+void indicator_config_save(void) {
+    eeprom_update_block(&g_value, ((void*)VIA_EEPROM_CUSTOM_CONFIG_ADDR), VIA_EEPROM_CUSTOM_CONFIG_SIZE);
+}
+
+void via_init_kb(void) {
+    if (via_eeprom_is_valid()) {
+        indicator_config_load();
+    } else {
+        indicator_config_save();
+    }
+}
+
+void via_custom_value_command_kb(uint8_t *data, uint8_t length) {
```

Comment:
Honestly, all of this should probably be handled at the keyboard level, not the keymap level.

Reply:
Not sure I follow. There's no other way to set which RGB LED gets triggered with what other than through the VIA custom menus, so I it belonged here.